src/ranked_match.hh

```diff
@@ -52,6 +53,7 @@ private:
     StringView m_candidate{};
     bool m_matches = false;
     Flags m_flags = Flags::None;
+    Vector<RankedMatch, MemoryDomain::RankedMatch> m_path_component_matches;
```

Comment:
If these semantics are okay we could get rid of this allocation by computing the path component matches lazily in `RankedMatch::operator<`. I think this could work fine because we use a heap to get the top 100 or so completions, so we never need to sort everything; the number of calls to `RankedMatch::operator<` is linear.
I'd need to check.